### PR TITLE
:zap: add UT for webhook

### PR DESCRIPTION
**Pull request by haitwang-cloud**
**Description:**

- Added unit tests for Webhook to improve code reliability and maintainability.
- Formatted Go code to improve code readability and consistency.
- Optimized code variable names to be more descriptive and meaningful.
- Optimized log messages to be more informative and clear.

### CODE REVIEW

**Comment by haitwang-cloud** [comment]
@archlitchi @lengrongfu @chaunceyjiang @calvin0327 @wawa0210 PTAL

**Comment by wawa0210** [comment]
@haitwang-cloud  thx, Code conflict, can you solve it ?

File: pkg/scheduler/webhook.go [hunk 1]
```diff
@@ -51,28 +51,30 @@ func (h *webhook) Handle(_ context.Context, req admission.Request) admission.Res
...
+		klog.Errorf("Failed to decode request: %v", err)
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 	if len(pod.Spec.Containers) == 0 {
+		klog.Warning("Denying admission as pod has no containers")
 		return admission.Denied("pod has no containers")
 	}
-	//klog.V(1).Infof("hook %v pod %v/%v", req.UID, req.Namespace, req.Name)
-	fmt.Printf("hook %v pod %v/%v", req.UID, req.Namespace, req.Name)
+	klog.Infof("Processing hook for pod %v/%v, UID: %v", req.Namespace, req.Name, req.UID)
```
Comment (wawa0210) [inline pkg/scheduler/webhook.go]: maybe use `klog.KObj(req)` would be more reasonable

**Comment by haitwang-cloud** [comment]
@wawa0210 PTAL, I have already addressed the comment and optimized the log using GPT-4.

**Pull request merged**